src/routers/alpha-router/gas-models/v3/v3-heuristic-gas-model.ts

```diff
@@ -367,7 +363,14 @@ export class V3HeuristicGasModelFactory extends IOnChainGasModelFactory {
     );
     const totalHops = BigNumber.from(routeWithValidQuote.route.pools.length);
 
-    const hopsGasUse = COST_PER_HOP(chainId).mul(totalHops);
+    let hopsGasUse = COST_PER_HOP(chainId).mul(totalHops);
+
+    // We have observed that this algorithm tends to underestimate single hop swaps.
+    // We add a buffer in the case of a single hop swap.
+    if (totalHops.eq(1)) {
+      hopsGasUse = hopsGasUse.add(SINGLE_HOP_OVERHEAD(chainId));
+    }
+
```

Comment:
Could we instead add the overhead to every calculation regardless of wether it is a single hop or not? the special case feel strange too.

Reply:
We've only observed the gas being underestimated for single hops, multiple hops seem much closer, so don't think we should apply it to everything